launcher/firstLaunch/firstlaunch_moc.cpp

```diff
@@ -279,7 +293,63 @@ QString FirstLaunchView::getHeroesInstallDir()
 	return QString{};
 }
 
-void FirstLaunchView::copyHeroesData(const QString & path)
+void FirstLaunchView::extractGogData()
+{
+#ifndef VCMI_NO_INNOEXTRACT
+	QString filterExe = tr("GOG executable") + " (*.exe)";
+	QString fileExe = QFileDialog::getOpenFileName(this, tr("Select a GOG installer (exe) file..."), QDir::homePath(), filterExe);
+	if(fileExe.isEmpty())
+		return;
+
+	QString filterBin = tr("GOG bin file") + " (*.bin)";
+	QString fileBin = QFileDialog::getOpenFileName(this, tr("Select a GOG data (bin) file..."), QFileInfo(fileExe).absolutePath(), filterBin);
+	if(fileBin.isEmpty())
+		return;
+
+	QTimer::singleShot(100, this, [this, fileExe, fileBin](){ // background to make sure FileDialog is closed...
```

Comment:
Why is this needed?

Reply:
It's a kind of workaround. The unpacking process blocks the UI.

Without it: On my testing system it doesn't even close the FileSelector until the process is done. I've tried to force update the UI before extracting, but it doesn't work. :(

With this workaround everything works great. The FileSelector is closed, the button changed name and color.

Reply:
> I've tried to force update the UI 

So calling qApp->processEvents() right before starting extraction doesn't help?

Anyway, extraction should be done on background thread (use QtConcurrent) and ui should show a spinner or indefinite progress bar (if the library doesn't provide progress updates).

Reply:
No. Tested it and it wouldn't help. :(

Yes. Would also prefer this. But my knowledge to QT is very limited now. The first project with it. So I have no idea how to do it the clean way.